open-telemetry/opentelemetry-collector-contrib
# [exporter/datasetexporter]: Call the currect function on shutdown

**Description:** Call the correct function on shutdown

After we have changed the underlying library we haven't updated the exporter code to call the correct function. This PR is fixing this.

**Link to tracking Issue:** #24253

**Testing:** 

1. Run open telemetry collector
2. Terminate it - `docker kill --signal SIGTERM ...`
3. Observe logs: 
```

2023-07-13T11:30:54.191Z	info	otelcol/collector.go:236	Received signal from OS	{"signal": "terminated"}
2023-07-13T11:30:54.191Z	info	service/service.go:157	Starting shutdown...
2023-07-13T11:30:54.191Z	info	adapter/receiver.go:139	Stopping stanza receiver	{"kind": "receiver", "name": "filelog/log", "data_type": "logs"}
...
...
2023-07-13T11:31:00.917Z	info	client/client.go:387	Buffers' Queue Stats:	{"kind": "exporter", "data_type": "logs", "name": "dataset/logs", "processed": 32, "enqueued": 32, "dropped": 0, "waiting": 0, "processingS": 24.727153498}
2023-07-13T11:31:00.919Z	info	client/client.go:399	Events' Queue Stats:	{"kind": "exporter", "data_type": "logs", "name": "dataset/logs", "processed": 200005, "enqueued": 200005, "waiting": 0, "processingS": 24.727153498}
2023-07-13T11:31:00.919Z	info	client/client.go:413	Transfer Stats:	{"kind": "exporter", "data_type": "logs", "name": "dataset/logs", "bytesSentMB": 150.06610584259033, "bytesAcceptedMB": 150.06610584259033, "throughputMBpS": 6.068879131385992, "perBufferMB": 4.689565807580948, "successRate": 1, "processingS": 24.727153498, "processing": 24.727153498}
2023-07-13T11:31:00.919Z	info	client/add_events.go:294	Finishing with success	{"kind": "exporter", "data_type": "logs", "name": "dataset/logs"}
2023-07-13T11:31:00.919Z	info	extensions/extensions.go:44	Stopping extensions...
2023-07-13T11:31:00.919Z	info	service/service.go:171	Shutdown complete.
```


**Documentation:** There is no change in the documentation needed.
